stdpopsim/utils.py

```diff
@@ -53,3 +54,24 @@ def is_valid_species_common_name(common_name):
     # FIXME any sensible restrictions we can make on common names? See #330.
     regex = re.compile(r"[A-Z].*")
     return regex.fullmatch(common_name) is not None
+
+
+def get_intervals_to_keep(mask_fpath, chrom):
+    """
+    Returns intervals to keep based on a bed file specified by the mask_fpath.
+    The mask must be in bed format (columns specify chrom, left, right) and
+    additional columns are ignored. The bed file specifies regions that are
+    to be kept, instead of deleted.
+    Intervals must be non-overlapping. The chromosome name must match
+    exactly ("22" is not equivalent to "chr22").
+    """
+    lines = np.loadtxt(
+        mask_fpath,
+        dtype={"names": ("chrom", "left", "right"), "formats": ("S15", "i4", "f4")},
```

Comment:
How about: `"formats": (object, int, int)`? If you use `object` here as the type for the chrom column, then you won't need to specify the string length, nor encode() the chrom parameter below. Also, it looks like you have different types for `left` and `right`.

Reply:
Heh, you're right, that's a bit of a mess. I think I was wrestling a lot with the `loadtxt` function, finally got something to work and didn't clean it up. I'll fix.